multicast/src/main/kotlin/com/dropbox/flow/multicast/ChannelManager.kt

```diff
@@ -261,7 +264,9 @@ internal class ChannelManager<T>(
                 }
             } else {
                 // unlock upstream since we now have a downstream that needs values
-                lastDeliveryAck?.complete(Unit)
+                if (!entry.piggybackOnly) {
```

Comment:
right now piggybacking downstreams also ack values. maybe we should change that as well or not do this.
Either way i think it should be consistent behavior.

Reply:
After thinking about it a bit more, I think we should actually ack from piggyback channels (both on creation and in carryover channel cases), which is what we currently do.

This is more consistent with thinking of the multiplexers as a server source of truth. When we don't have a persister, we subscribe to changes from the network and continue to listen to them.

Thoughts?

Reply:
There is one tricky case with infinite fetchers.
If developer issues a cahed, refresh=false request with the intent of just observing new values w/o triggering a fetch, they will end up triggering the consumption of the stream hence more fetch emissions (if fetcher is a flow). 
That being said,in that case if cache is empty, they will already trigger infinite consumption of the fetcher stream

Reply:
Maybe the right solution for that case is to have a request type that would never trigger the fetcher

Reply:
> There is one tricky case with infinite fetchers.
> If developer issues a cahed, refresh=false request with the intent of just observing new values w/o triggering a fetch, they will end up triggering the consumption of the stream hence more fetch emissions (if fetcher is a flow).
> That being said,in that case if cache is empty, they will already trigger infinite consumption of the fetcher stream

In order for the stream to be started, you would need to have at least one one-non piggyback request. So the actual edge case would be: add piggyback (nothing happens). add regular channel which will trigger the infinite fetcher (multicasted to together with the piggyback channel). regular channel is manually closed. infinite fetcher still runs because piggyback unblocks it.

That being said, the way for this edge case to manifest is for a user to asked for a cached value, get a hit and still affect the fetcher. But the user doesn't know if the cache would be hot or not so there's no expectation that the call won't affect the network. I know this doesn't solve the problem in the multicaster itself, but at least this behavior doesn't create a bug in Store.

If we wanted to solve this issue I think we should make the piggyback behavior consistent between manual piggyback and the current non-manual piggybacks (as in non-manual piggyback should also not ack the network). That would probably be better done in another PR though as it's a bigger change.